src/classes/CON_Preferred_Email_TEST.cls

```diff
@@ -0,0 +1,2215 @@
+        System.assertEquals('test@sf.org', insertedContactList[1].Email);
+        System.assertEquals('Alternate Email', insertedContactList[1].Preferred_Email__c);
+
+        insertedContactList[0].AlternateEmail__c = '';
+        insertedContactList[0].Preferred_Email__c = null;
+        insertedContactList[1].AlternateEmail__c = '';
+        insertedContactList[1].Preferred_Email__c = null;
+        
+        Test.startTest();
+            update insertedContactList;
+        Test.stopTest();
+
+        List<Contact> updatedContactList = [SELECT Id, AlternateEmail__c, Preferred_Email__c, Email FROM Contact];
+
+        System.assertEquals(null, updatedContactList[0].AlternateEmail__c, 'Alternate Email should be cleared');
+        System.assertEquals(null, updatedContactList[0].Preferred_Email__c, 'Preferred Email should be cleared');
+        System.assertEquals(null, updatedContactList[0].Email, 'Standard Email should be cleared');
+        System.assertEquals(null, updatedContactList[1].AlternateEmail__c, 'Alternate Email should be cleared');
+        System.assertEquals(null, updatedContactList[1].Preferred_Email__c, 'Preferred Email should be cleared');
+        System.assertEquals(null, updatedContactList[1].Email, 'Standard Email should be cleared');
+    } 
+    
+    /*********************************************************************************************************************************
+    * @description Tests Updating a Contact to clear Preferred Email and all other custom email fields will not clear the value in
+    * standard email field
+    *********************************************************************************************************************************/
+    @isTest 
+    static void clearEmailFieldsOnUpdateNegative() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Disable_Preferred_Email_Enforcement__c = false)
+            );
+
+        List<Contact> testContactList = UTIL_UnitTestData_TEST.getMultipleTestContacts(2);
+        testContactList[0].WorkEmail__c = '';
+        testContactList[0].Preferred_Email__c = '';
+        testContactList[0].AlternateEmail__c = 'test@sf.org';
+        testContactList[1].WorkEmail__c = '';
+        testContactList[1].Preferred_Email__c = '';
+        testContactList[1].AlternateEmail__c = 'test@sf.org';
+        insert testContactList;
+
+        List<Contact> insertedContactList = [SELECT Id, Email, AlternateEmail__c, Preferred_Email__c FROM Contact];
+
+        System.assertEquals('test@sf.org', insertedContactList[0].Email);
+        System.assertEquals('Alternate Email', insertedContactList[0].Preferred_Email__c);
+        System.assertEquals('test@sf.org', insertedContactList[1].Email);
+        System.assertEquals('Alternate Email', insertedContactList[1].Preferred_Email__c);
+
+        MAPR_CON_PreferredEmailFields.preferredEmailFieldSettingsModel = null;
```

Comment:
@MatthewBlanski wanted to check this with you, clearing the cache here so that new values are retrieved for the line below.